sound/soc/sof/ipc4-topology.c

```diff
@@ -1431,13 +1431,58 @@ static int sof_ipc4_control_load_volume(struct snd_sof_dev *sdev, struct snd_sof
 	return 0;
 }
 
+static int sof_ipc4_control_load_bytes(struct snd_sof_dev *sdev, struct snd_sof_control *scontrol)
+{
+	struct sof_ipc4_control_data *control_data;
+	int ret;
+
+	scontrol->size = sizeof(struct sof_ipc4_control_data) + scontrol->priv_size;
+
+	scontrol->ipc_control_data = kzalloc(scontrol->size, GFP_KERNEL);
+	if (!scontrol->ipc_control_data)
+		return -ENOMEM;
+
+	control_data = scontrol->ipc_control_data;
+	control_data->index = scontrol->index;
+	if (scontrol->priv_size > 0) {
+		memcpy(control_data->data, scontrol->priv, scontrol->priv_size);
+
+		if (control_data->data->magic != SOF_ABI_MAGIC) {
```

Comment:
Where is `control_data->data->magic` assigned? Do we really need to check `magic`?

Reply:
> Where is `control_data->data->magic` assigned? Do we really need to check `magic`?

This is assigned in topology. This is to make sure the kcontrol data is for SOF IPC, just for the sanity check.